### ISSUE TITLE
Description of some details in the PrecalcConstraint

### ISSUE THREAD

**Issue by lj-at-work** [mitre/im_sample_algorithm#4]
Hi, I'm working my way through the sample algorithm and I think it's going well, but I'd like to ask for clarification on a few of the fields in `PrecalcConstraint` in particular,

```c++
   ActiveFlagType active_flag;
   bool violation_flag;
```

I'm inferring these fields are related to the initial trajectory prediction, since they are read in the `KinematicDescent4DPredictor`, but I was wondering if there was anywhere that elaborates on the particular meanings of the enum/flags. 

I didn't find writes beyond copy/initialization, so I'm assuming what is doing the writes for you is redacted. Is there a place that describes what ought to be done here that I just over-looked?

**Comment by sbowman-mitre** [comment]
> I'm inferring these fields are related to the initial trajectory prediction

Yup, correct.

Hrm...I thought this constraint logic was discussed in DO-361A Appendix C (published earlier this year), but I just reviewed it and don't see what I was remembering. I know we had a draft review that ended up removing a bunch of workflows & this discussion must have been removed during that process. Bummer & sorry about that. But, there is a bread crumb in the FMACM code. See [PrecalcConstraint.h](https://github.com/mitre/FMACM/blob/3c81f575e4de73d83c02c0ea3b379a726d91837a/include/public/PrecalcConstraint.h#L21):
```c++
enum class ActiveFlagType
{
   UNSET = 0,
   BELOW_ALT_ON_SPEED = 1,
   AT_ALT_ON_SPEED,
   BELOW_ALT_SLOW,
   SEG_END_LOW_ALT,
   SEG_END_MID_ALT,
   AT_ALT_SLOW,
   SEG_END_AT_ALT
};
```

As the prediction algorithm cycles through the trajectory segments defined in C.2.2.1, it defaults to using the algorithm in C.2.2.1.1. At the end of each segment (or even more regularly), the algorithm checks to make sure the current constraint (altitude and/or speed) is not violated. These flags are used to define the kind of constraint violations that can occur. The specific kind of violation is then used to determine which algorithm in C.2.2.1 is best for resolving the conflict. Does that help?
